### PR TITLE
Fixed GPU-CPU tensor conversion bug. Fixed imports. Added compatibility with sklearn 0.20.1

### PR DESCRIPTION
**Pull request by hlibt**
I am not sure if it is support of sklearn 0.20.1+ is desirable. However, changes should not break backward compatibility with sklearn 0.20.0.

### CODE REVIEW

File: combine_ucr.py [hunk 1]
```diff
@@ -135,13 +136,13 @@ def parse_arguments():
                 )
                 grid_search.fit(split[0], split[2])
             classifier = grid_search.best_estimator_
-        sklearn.externals.joblib.dump(
+        joblib.dump(
```
Comment (White-Link) [inline combine_ucr.py]: I would prefer keeping the sklearn.externals version of joblib. With your code it seems impossible to load my pretrained models.

File: scikit_wrappers.py [hunk 2]
```diff
@@ -321,7 +324,7 @@ def encode(self, X, batch_size=50):
                         batch = batch.cuda(self.gpu)
                     features[
                         count * batch_size: (count + 1) * batch_size
-                    ] = self.encoder(batch)
+                    ] = self.encoder(batch).cpu()
```
Comment (White-Link) [inline scikit_wrappers.py:326]: Could you add the same .cpu() call on line 338 in case all time series do not have equal lengths?

**Review by White-Link — CHANGES_REQUESTED** [review]
Thanks for the suggestions!

Overall I would want to keep using sklearn.externals.joblib to keep compatibility with my pretrained models. Please refer to my comments.

File: combine_ucr.py (hunk 1 above)
Comment (hlibt) [reply combine_ucr.py]: I have reverted changes regarding joblib. However, it is very strange that you cannot load you the pretrained models. I have checked and haven't discovered any changes in the default serialization protocols.  Could you please provide me an example of a pretrained model?

Comment (White-Link) [reply combine_ucr.py]: Hi, sorry for the very late reply! I added an example of a pretrained model in the repository in case you want to check.

**Pull request merged**